src/Api/Serializer/Ec2ParamBuilder.php

```diff
@@ -12,7 +12,7 @@ class Ec2ParamBuilder extends QueryParamBuilder
     protected function queryName(Shape $shape, $default = null)
     {
         return ($shape['queryName']
-            ?: ucfirst($shape['locationName']))
+            ?: (isset($shape['locationName']) ? ucfirst($shape['locationName']) : ""))
```

Comment:
Sorry for giving mixed feedback here a few times; this could be shortenened to 
`ucfirst($shape['locationName'] ?: "")`